zipkin-storage/elasticsearch-http/src/main/java/zipkin/storage/elasticsearch/jest/RestClient.java

```diff
@@ -0,0 +1,397 @@
+
+        ImmutableList.Builder<Span> builder = ImmutableList.builder();
+        for (SearchResult.Hit<Span, ?> hit : input.getHits(Span.class)) {
+          builder.add(hit.source);
+        }
+        return builder.build();
+      }
+    });
+  }
+
+  @Override
+  public ListenableFuture<Collection<DependencyLink>> findDependencies(String[] indices) {
+    QueryBuilder query = QueryBuilders.matchAllQuery();
+    if (indices.length > MAX_INDICES) {
+      query = QueryBuilders.indicesQuery(query, indices).noMatchQuery("none");
+      indices = allIndices;
+    }
+
+    Search.Builder search = new Search.Builder(new SearchSourceBuilder().query(query).toString())
+        .addIndex(Arrays.asList(indices))
+        .addType(DEPENDENCY_LINK);
+
+    return transform(toGuava(new LenientSearch(search)),
+        new Function<SearchResult, Collection<DependencyLink>>() {
+          @Override public Collection<DependencyLink> apply(SearchResult input) {
+            ImmutableList.Builder<DependencyLink> builder = ImmutableList.builder();
+            for (SearchResult.Hit<DependencyLink, ?> hit : input.getHits(DependencyLink.class)) {
+              builder.add(hit.source);
+            }
+            return builder.build();
+          }
+        });
+  }
+
+  @Override
+  public ListenableFuture<Void> indexSpans(List<IndexableSpan> spans) {
+    if (spans.isEmpty()) return Futures.immediateFuture(null);
+
+    // Create a bulk request when there is more than one span to store
+    ListenableFuture<?> future;
+    final Set<String> indices = new HashSet<>();
+    if (spans.size() == 1) {
+      IndexableSpan span = getOnlyElement(spans);
+      future = toGuava(toIndexRequest(span));
+      if (flushOnWrites) {
+        indices.add(span.index);
+      }
+    } else {
+      Bulk.Builder batch = new Bulk.Builder();
+      for (IndexableSpan span : spans) {
+        batch.addAction(toIndexRequest(span));
+        if (flushOnWrites) {
+          indices.add(span.index);
+        }
+      }
+      future = toGuava(batch.build());
+    }
+
+    if (flushOnWrites) {
+      future = transform(future, new AsyncFunction() {
+        @Override public ListenableFuture apply(Object input) {
+          return toGuava(new Flush.Builder().addIndex(indices).build());
+        }
+      });
+    }
+
+    return transform(future, Functions.<Void>constant(null));
+  }
+
+  @Override protected void ensureClusterReady(final String catchAll) {
+    class IndicesHealth extends Health {
+      private IndicesHealth(Builder builder) {
+        super(builder);
+      }
+
+      @Override
+      protected String buildURI() {
+        return super.buildURI() + catchAll;
+      }
+    }
+
+    String status = executeUnchecked(new IndicesHealth(new Health.Builder()))
+        .getJsonObject().get("status").getAsString();
+    checkState(!"RED".equalsIgnoreCase(status), "Health status is RED");
+  }
+
+  private Index toIndexRequest(IndexableSpan span) {
```

Comment:
static utility methods unless they require access to the enclosing object. makes it easier to see impact.